DependencyInjection/HttplugExtension.php

```diff
@@ -198,14 +208,29 @@ private function configureAuthentication(ContainerBuilder $container, array $con
      * @param ContainerBuilder $container
      * @param string           $name
      * @param array            $arguments
-     *
-     * @return string The service id of the client.
+     * @param bool             $profiling
      */
-    private function configureClient(ContainerBuilder $container, $name, array $arguments)
+    private function configureClient(ContainerBuilder $container, $name, array $arguments, $profiling)
     {
         $serviceId = 'httplug.client.'.$name;
-        $definition = $container->register($serviceId, DummyClient::class);
-        $definition->setFactory([PluginClientFactory::class, 'createPluginClient'])
+
+        $clientOptions = [];
```

Comment:
I would name this $pluginClientOptions just to be clear. The 3rd parameter to createPluginClient is actually options to the client. 

https://github.com/php-http/HttplugBundle/blob/dev-issue95/ClientFactory/PluginClientFactory.php#L18-L19

(same with the registerAutoDiscoverableClient())